src/storage/src/healthcheck.rs

```diff
@@ -348,22 +588,17 @@ where
                     let new_round = outputs_seen.insert(output_index);
 
                     if !is_active_worker {
-                        warn!(
+                        error!(
                             "Health messages for {object_type} {id} passed to \
                               an unexpected worker id: {healthcheck_worker_id}"
                         )
                     }
 
                     if health_event.should_halt() {
-                        *halt_with = Some(health_event.clone());
+                        *halt_with = Some((ns.clone(), health_event.clone()));
                     }
 
-                    let update = Some(health_event);
-                    // Keep the max of the messages in each round; this ensures that errors don't
-                    // get lost while also letting us frequently update to the newest status.
-                    if new_round || &healths[worker_id] < &update {
-                        healths[worker_id] = update;
-                    }
+                    healths.merge_update(worker_id, ns, health_event, !new_round);
```

Comment:
Can you talk me through the "round" think? Seems weird that the resulting status would depend on how timely decides to batch stuff up...

Reply:
Its old old code; its basically just to ensure we pick the worst status that came in a single timely buffer for a worker deterministically